kafka-rest-common/src/main/java/io/confluent/kafkarest/converters/AvroConverter.java

```diff
@@ -18,76 +18,40 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.google.common.annotations.VisibleForTesting;
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.avro.AvroSchema;
+import io.confluent.kafka.schemaregistry.avro.AvroSchemaUtils;
 import org.apache.avro.Schema;
-import org.apache.avro.generic.GenericContainer;
-import org.apache.avro.generic.GenericDatumReader;
-import org.apache.avro.generic.GenericDatumWriter;
-import org.apache.avro.io.DatumReader;
-import org.apache.avro.io.DatumWriter;
-import org.apache.avro.io.DecoderFactory;
-import org.apache.avro.io.EncoderFactory;
-import org.apache.avro.io.JsonEncoder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.nio.ByteBuffer;
-import java.util.HashMap;
-import java.util.Map;
 
 /**
  * Provides conversion of JSON to/from Avro.
  */
-public class AvroConverter {
+public class AvroConverter implements SchemaConverter {
 
   private static final Logger log = LoggerFactory.getLogger(AvroConverter.class);
 
-  private static final EncoderFactory encoderFactory = EncoderFactory.get();
-  private static final DecoderFactory decoderFactory = DecoderFactory.get();
   private static final ObjectMapper jsonMapper = new ObjectMapper();
 
-  private static final Map<String, Schema> primitiveSchemas;
-
-  static {
-    Schema.Parser parser = new Schema.Parser();
-    primitiveSchemas = new HashMap<String, Schema>();
-    primitiveSchemas.put("Null", createPrimitiveSchema(parser, "null"));
-    primitiveSchemas.put("Boolean", createPrimitiveSchema(parser, "boolean"));
-    primitiveSchemas.put("Integer", createPrimitiveSchema(parser, "int"));
-    primitiveSchemas.put("Long", createPrimitiveSchema(parser, "long"));
-    primitiveSchemas.put("Float", createPrimitiveSchema(parser, "float"));
-    primitiveSchemas.put("Double", createPrimitiveSchema(parser, "double"));
-    primitiveSchemas.put("String", createPrimitiveSchema(parser, "string"));
-    primitiveSchemas.put("Bytes", createPrimitiveSchema(parser, "bytes"));
-  }
-
-  public static Object toAvro(JsonNode value, Schema schema) {
+  @Override
+  public Object toObject(JsonNode value, ParsedSchema parsedSchema) {
     try {
-      ByteArrayOutputStream out = new ByteArrayOutputStream();
-      jsonMapper.writeValue(out, value);
-      DatumReader<Object> reader = new GenericDatumReader<Object>(schema);
-      Object object = reader.read(
-          null, decoderFactory.jsonDecoder(schema, new ByteArrayInputStream(out.toByteArray())));
-      out.close();
-      return object;
-    } catch (IOException | RuntimeException e) {
+      return AvroSchemaUtils.toObject(value, (AvroSchema) parsedSchema);
```

Comment:
Either rename AvroSchemaUtils to AvroUtils, and ProtobufSchemaUtils to ProtobufUtils, or rename JsonSchemaUtils to JsonSchemaSchemaUtils.

Reply:
These classes are in Schema Registry and match the `AvroSchema`, `ProtobufSchema`, and `JsonSchema` classes.  I think I'll leave the current naming scheme for now.